### PR TITLE
Improve the coding style according to OPStack's guide

### PR DESCRIPTION
**Pull request by qzhodl**
The coding style guide is [here](https://github.com/ethstorage/optimism/blob/develop/packages/contracts-bedrock/STYLE_GUIDE.md)

The changes are mainly for:
 - Natspec comments style for all the files that we write
 - Naming conventions for constants, immutables, and parameters
 - Error handling messages
 - Restructuring the file folders
 
The PR has passed the UI test.

### CODE REVIEW

File: contracts/EthStorageContract.sol [hunk 1]
```diff
@@ -2,27 +2,47 @@
...
 
+    /// @notice Constructs the EthStorageContract contract.
     constructor(
         Config memory _config,
         uint256 _startTime,
         uint256 _storageCost,
         uint256 _dcfFactor
-    ) StorageContract(_config, _startTime, _storageCost, _dcfFactor) {}
+    ) StorageContract(_config, _startTime, _storageCost, _dcfFactor) {
+        //initialize(0, 0, 0, address(0x0), msg.sender);
```
Comment (iteyelmp) [inline contracts/EthStorageContract.sol:42]: Should it be deleted? The constructor should not call this function.

Comment (qzhodl) [reply contracts/EthStorageContract.sol:42]: You can check the guide here: https://github.com/ethereum-optimism/optimism/blob/457f33f4fdda9373dcf2839619ebf67182ee5057/packages/contracts-bedrock/STYLE_GUIDE.md?plain=1#L109

File: contracts/DecentralizedKV.sol [hunk 2]
```diff
@@ -2,94 +2,125 @@
...
 
-    /* skey - PhyAddr */
+    /// @notice skey and PhyAddr mapping
     mapping(bytes32 => PhyAddr) internal kvMap;
-    /* index - skey, reverse lookup */
+
+    /// @notice index and skey mapping, reverse lookup
     mapping(uint256 => bytes32) internal idxMap;
 
+    /// @notice Emitted when an key-value is removed.
```
Comment (syntrust) [inline contracts/DecentralizedKV.sol:58]: a key-value

File: contracts/EthStorageContract2.sol [hunk 3]
```diff
@@ -2,100 +2,133 @@
...
     constructor(
         Config memory _config,
         uint256 _startTime,
         uint256 _storageCost,
         uint256 _dcfFactor
     ) EthStorageContract(_config, _startTime, _storageCost, _dcfFactor) {}
 
-    function getEncodingKey(uint256 kvIdx, address miner) internal view returns (uint256) {
-        return uint256(keccak256(abi.encode(kvMap[idxMap[kvIdx]].hash, miner, kvIdx))) % modulusBn254;
+    /// @notice Comupte the encoding key using the kvIdx and miner address
```
Comment (syntrust) [inline contracts/EthStorageContract2.sol:19]: typo of Compute

File: contracts/EthStorageContract2.sol [hunk 4]
```diff
@@ -2,100 +2,133 @@
...
-        (uint256 kvIdx, uint256 sampleIdxInKv) = getSampleIdx(rows, startShardId, hash0);
-                
+        (uint256 kvIdx, uint256 sampleIdxInKv) = getSampleIdx(_rows, _startShardId, _hash0);
+
         PhyAddr memory kvInfo = kvMap[idxMap[kvIdx]];
 
         require(
-            checkInclusive(kvInfo.hash, sampleIdxInKv, mask ^ uint256(encodedSample), inclusiveProof),
+            checkInclusive(kvInfo.hash, sampleIdxInKv, _mask ^ uint256(_encodedSample), _inclusiveProof),
             "invalid samples"
```
Comment (syntrust) [inline contracts/EthStorageContract2.sol:95]: "EthStorageContract2: invalid samples"

File: contracts/EthStorageContract2.sol [hunk 5]
```diff
@@ -2,100 +2,133 @@
...
+        bytes32[] memory _encodedSamples,
+        uint256[] memory _masks,
+        bytes[] calldata _inclusiveProofs,
+        bytes[] calldata _decodeProof
     ) public view virtual override returns (bytes32) {
-        require(encodedSamples.length == randomChecks, "data length mismatch");
-        require(masks.length == randomChecks, "masks length mismatch");
-        require(inclusiveProofs.length == randomChecks, "proof length mismatch");
-        require(decodeProof.length == 1, "decodeProof length mismatch");
+        require(_encodedSamples.length == RANDOM_CHECKS, "data length mismatch");
```
Comment (syntrust) [inline contracts/EthStorageContract2.sol:111]: "EthStorageContract2: {message}" for all requires?

File: contracts/EthStorageContractL2.sol [hunk 6]
```diff
@@ -3,65 +3,87 @@ pragma solidity ^0.8.0;
...
+        uint256 _nonce,
+        bytes32[] memory _encodedSamples,
+        uint256[] memory _masks,
+        bytes calldata _randaoProof,
+        bytes[] calldata _inclusiveProofs,
+        bytes[] calldata _decodeProof
     ) internal override {
         // Obtain the blockhash of the block number of recent blocks
-        require(l1Block.number() - blockNumber <= maxL1MiningDrift, "block number too old");
+        require(L1_BLOCK.number() - _blockNumber <= MAX_L1_MINING_DRIFT, "block number too old");
```
Comment (syntrust) [inline contracts/EthStorageContractL2.sol:68]: "EthStorageContractL2: block number too old"

File: contracts/EthStorageContractL2.sol [hunk 7]
```diff
@@ -3,65 +3,87 @@ pragma solidity ^0.8.0;
...
 
-        bytes32 hash0 = getRandao(blockNumber, randaoProof);
+        bytes32 hash0 = _getRandao(_blockNumber, _randaoProof);
         // Estimate block timestamp
-        uint256 mineTs = l1Block.timestamp() - (l1Block.number() - blockNumber) * 12;
+        uint256 mineTs = L1_BLOCK.timestamp() - (L1_BLOCK.number() - _blockNumber) * 12;
 
         // Given a blockhash and a miner, we only allow sampling up to nonce limit times.
-        require(nonce < nonceLimit, "nonce too big");
+        require(_nonce < nonceLimit, "nonce too big");
```
Comment (syntrust) [inline contracts/EthStorageContractL2.sol:76]: "EthStorageContractL2: nonce too big"

File: contracts/libraries/RandaoLib.sol [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+/// @notice Handles Randao related operations
+library RandaoLib {
+    using RLPReader for RLPReader.RLPItem;
+    using RLPReader for RLPReader.Iterator;
+    using RLPReader for bytes;
+
+    /// @notice Get the Randao mixDigest from the header
+    /// @param _item The RLP data of the header
+    /// @return The Randao mixDigest
+    function getRandaoFromHeader(RLPReader.RLPItem memory _item) internal pure returns (bytes32) {
```
Comment (syntrust) [inline contracts/libraries/RandaoLib.sol:16]: Is it applied here that internal functions are prefixed with an underscore?

File: contracts/DecentralizedKV.sol [hunk 9]
```diff
@@ -2,94 +2,125 @@
...
+    /// @notice Maximum size of a single key-value pair
+    uint256 internal immutable MAX_KV_SIZE;
+
+    /// @custom:legacy
     /// @custom:spacer storageCost, dcfFactor, startTime, maxKvSize
     /// @notice Spacer for backwards compatibility.
-    uint256[4] public kvSpacers;
-
-    uint40 public kvEntryCount; // number of entries in the store
+    uint256[4] private kvSpacers;
```
Comment (syntrust) [inline contracts/DecentralizedKV.sol:47]: Is it necessary to apply the spacer naming rule?
> `The name of a spacer variable MUST be in the format spacer_<slot>_<offset>_<length> where <slot> is the original storage slot number, <offset> is the original offset position within the storage slot, and <length> is the original size of the variable.`

File: contracts/DecentralizedKV.sol [hunk 10]
```diff
@@ -2,94 +2,125 @@
...
+    /// @param _owner The contract owner.
+    function __init_KV(address _owner) internal onlyInitializing {
         __Context_init();
         __Ownable_init(_owner);
         kvEntryCount = 0;
     }
 
-    function pow(uint256 fp, uint256 n) internal pure returns (uint256) {
-        return BinaryRelated.pow(fp, n);
+    /// @notice pow function in Q128.
```
Comment (iteyelmp) [inline contracts/DecentralizedKV.sol:81]: Pow

File: contracts/DecentralizedKV.sol (hunk 9 above)
Comment (qzhodl) [reply contracts/DecentralizedKV.sol:47]: Those spacers are temporary and will be deleted in the next version, so we are just using an array to represent them. After the contract is live on mainnet, we’ll follow the same practices as the code style.

File: contracts/EthStorageContract.sol [hunk 11]
```diff
@@ -33,7 +53,12 @@ contract EthStorageContract is StorageContract, Decoder {
         __init_storage(_minimumDiff, _prepaidAmount, _nonceLimit, _treasury, _owner);
     }
 
-    function modExp(uint256 _b, uint256 _e, uint256 _m) internal view returns (uint256 result) {
+    /// @notice Performs modular exponentiation, which is a type of exponentiation performed over a modulus.
+    /// @param _b The base
+    /// @param _e The exponent
+    /// @param _m The modulus
+    /// @return result The result of the modular exponentiation
+    function _modExp(uint256 _b, uint256 _e, uint256 _m) internal view returns (uint256 result) {
```
Comment (iteyelmp) [inline contracts/EthStorageContract.sol:61]: returns (uint256 result_) ?

File: contracts/libraries/RandaoLib.sol (hunk 8 above)
Comment (qzhodl) [reply contracts/libraries/RandaoLib.sol:16]: According to the libraries in opstack, there is no underscore prefixed on the internal functions

**Review by iteyelmp — APPROVED** [review]

File: contracts/EthStorageContract2.sol [hunk 12]
```diff
@@ -2,100 +2,133 @@
...
+                _hash0,
+                _encodedSamples[i],
+                _masks[i],
+                _inclusiveProofs[i]
             );
         }
 
-        require(decodeSample(masks, kvIdxs, sampleIdxs, miner, decodeProof[0]), "decode failed");
-        return hash0;
+        require(decodeSample(_masks, kvIdxs, sampleIdxs, _miner, _decodeProof[0]), "decode failed");
```
Comment (syntrust) [inline contracts/EthStorageContract2.sol:131]: contract name missing in `require`

**Review by syntrust — APPROVED** [review]

**Pull request merged**